lib/pay/payments.ex

```diff
@@ -488,6 +488,25 @@ defmodule Pay.Payments do
     end
   end
 
+  @doc """
+  Returns the list of payment_events by the given payment external_id.
+
+  ## Examples
+
+      iex> list_payment_events_by_payment_external_id("3bfd1a3c-0960-49da-be66-053b159df62e")
+      [%PaymentEvent{}, ...]
+
+  """
+  @spec list_payment_events_by_payment_external_id(String.t()) :: [%PaymentEvent{}]
+  def list_payment_events_by_payment_external_id(external_id) do
```

Comment:
compose what we've already got. can just do it in the controller

`payment_id |> get_payment_by_external_id! |> list_payment_events`

Reply:
@dagoof alright I'll give it a go. i had a quick look at the `from()` query but it didn't seem possible to compose what we already had. i'll take another look